spec/integration/metric_spec.rb

```diff
@@ -7,489 +7,524 @@
 # time constants
 t4h = 4 * 60 * 60 * 1000
 
-describe 'Simple', :vcr do
-  it 'Should be Cool' do
-    Net::HTTP.get_response(URI('http://localhost:8080/'))
-  end
-end
+# test contexts
+v8_context = :metrics_0_8_0
+services_context = :metrics_services
 
-describe 'Tenants', vcr: { match_requests_on: [:uri, :method], record: :none } do
-  before(:all) do
-    setup_client
+[v8_context, services_context].each do |metrics_context|
+  if ENV['SKIP_V8_METRICS'] == '1' && metrics_context == v8_context
+    puts 'skipping v8 metrics'
+    next
   end
-
-  it 'Should create and return tenant' do
-    tenant = SecureRandom.uuid
-    @client.tenants.create(tenant)
-    created = @client.tenants.query.select { |t| t.id == tenant }
-    expect(created).not_to be nil
+  if ENV['SKIP_SERVICES_METRICS'] == '1' && metrics_context == services_context
+    puts 'skipping services metrics'
+    next
   end
-end
-
-describe 'No_Tenant' do
-  it 'Should fail' do
-    id = SecureRandom.uuid
-
-    VCR.use_cassette('No_Tenant/Should fail', erb: { id: id }, record: :none) do
-      setup_client
 
-      begin
-        @client.counters.push_data(id, value: 4)
-      rescue # rubocop:disable Lint/HandleExceptions
-        # This is good
+  describe "#{metrics_context}" do
+    let(:cassette_name) do |example|
+      if example.respond_to?(:example_group) && !example.example_group.description.start_with?(metrics_context.to_s)
+        example.example_group.description + '/'
       else
-        fail 'The call should have failed due to missing tenant'
-      end
+        ''
+      end + example.description
     end
-  end
-end
-
-describe 'Mixed metrics' do
-  before(:all) do
-    setup_client_new_tenant
-  end
-
-  it 'Should send mixed metric request of a single type' do
-    id = SecureRandom.uuid
 
-    VCR.use_cassette('Mixed_metrics/Should send mixed metric request of a single type',
-                     erb: { id: id }, record: :none
-                    ) do
-      @client.push_data(counters: [{ id: id, data: [{ value: 1 }] }])
-      data = @client.counters.get_data(id)
-      expect(data.size).to be 1
-
-      @client.push_data(availabilities: [{ id: id, data: [{ value: 'down' }] }])
-      data = @client.avail.get_data(id)
-      expect(data.size).to be 1
-
-      @client.push_data(gauges: [{ id: id, data: [{ value: 1.1 }] }])
-      data = @client.gauges.get_data(id)
-      expect(data.size).to be 1
+    around(:each) do |example|
+      run_for = example.metadata[:run_for]
+      if run_for.nil? || run_for.empty? || run_for.include?(metrics_context)
+        @random_id = SecureRandom.uuid
+        if example.metadata[:skip_auto_vcr]
+          example.run
+        else
+          record("Metrics/#{metrics_context}", { id: @random_id }, cassette_name, example: example)
```

Comment:
a helper (https://git.io/vKmoW)

Good thing about it is that the old tests had to repeat the name and all the previous contexts in the string that was used for `VCR.use_cassette` (like here: https://git.io/vKmoQ), this way, it's auto-inferred.